### PR TITLE
Fixed terra-status-view issues

### PR DESCRIPTION
**Pull request by PK106552**
### Summary
UXPLATFORM-8840 - Terra-Status-View- Enable screenreader announce dynamic variant details.

### Issue:
1. Screenreader not  auto reading dynamic variant details. 
### Solution:
If we have dynamic variant examples we are including aria-live=polite and role="alert" attribute so as soon any action is triggered screenreader will automatically read dynamic variant details.

### Change variant dropdown example (VO):
1. Navigate to variant dropdown using tab key.
5. Open dropdown list using Ctrl+Option+Space.
6. Select variant name using Ctrl+Option+Space.
7. Selected variant details will be displayed and VO read selected variant details.

### Toggle variant example (VO):
1. Navigate to toggle variant button using tab key.
2. To select toggle variant button press Ctrl+Option+Space.
3. Variant details will be displayed and VO read selected variant details.

### Change variant dropdown example (JAWS):
1. Navigate to variant dropdown using tab key.
2. Select variant name  list using arrow key.
3. JAWS read selected variant details.

### Toggle variant example (JAWS):
1. Navigate to toggle variant button using tab key.
2. To select toggle variant button press Enter.
4. Variant details will be displayed and JAWS read selected variant details.

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/status-view/example/Example.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import StatusView from 'terra-status-view';
+import ToggleButton from 'terra-toggle-button';
+
+const Example = () => (
+  <ToggleButton closedButtonText="ToggleButton">
+    <StatusView
+      role="alert"
+      aria-live="polite"
+      message="Testing"
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/status-view/example/Example.jsx:10]:
```suggestion
      message="Testing"
```
Update this with meaning full text

File: packages/terra-core-docs/src/terra-dev-site/doc/status-view/example/Custom.jsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ const StatusViewButtons = [
 const Custom = () => (
   <StatusView
     message="Custom Message"
+    aria-live="polite"
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/status-view/example/Custom.jsx:18]: Let's consider an scenario I am having terra-status-view in my project named `projectA` and `projectB` so do you expect me to add aria-live for all the status-view added in `projectA` and `projectB` to make it accessible for screen readers ??

**Comment by supreethmr** [comment]
> ### Summary
> UXPLATFORM-8840 - Terra-Status-View- Enable focusing and announce variant details
> 
> ### Solution
> Added Attributes to announce variant name name message.

What are steps you used for testing terra-status-view changes ??

**Comment by sdadn** [comment]
Adding the `do not merge` label since there are additional considerations needed for the PR.

File: packages/terra-core-docs/src/terra-dev-site/doc/status-view/About.1.doc.mdx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { Badge } from 'terra-status-view/package.json?dev-site-package';
 import ToggleVariants from './example/ToggleVariants?dev-site-example';
 import ToggleAlignmentAndGlyph from './example/ToggleAlignmentAndGlyph?dev-site-example';
 import Custom from './example/Custom?dev-site-example';
+import Example from './example/Example?dev-site-example'
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/status-view/About.1.doc.mdx:6]:
```suggestion
import DynamicExample from './example/Example?dev-site-example'
```

File: packages/terra-core-docs/src/terra-dev-site/doc/status-view/About.1.doc.mdx [hunk 4]
```diff
@@ -51,6 +52,7 @@ Set `focusable=false` for svg element used for `customGlyph` prop, to prevent fo
 <ToggleVariants title="Variants" />
 <Custom title="Custom: Icon + message + buttons" />
 <ToggleAlignmentAndGlyph />
+<Example title="Dynamic Status View" />
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/status-view/About.1.doc.mdx:55]:
```suggestion
<ToggleStatusView title="Toggle Status View" />
```

**Comment by sdadn** [comment]
Can the PR Description/Jira be updated with a clear description of the issue and acceptance criteria? It is difficult to review/approve this PR without a clear understanding of the goals.

**Comment by PK106552** [comment]
> PR Description

Hi @sdadn ,  I have updated the PR description about this issue and solution

File: packages/terra-core-docs/CHANGELOG.md [hunk 5]
```diff
@@ -3,12 +3,16 @@
 ## Unreleased
 
 * Added
+
+  * `aria-live="polite"` & `alert="role"` added to auto announce variant details.
```
Comment (supreethmr) [inline packages/terra-core-docs/CHANGELOG.md:7]:
```suggestion
  * Added dynamic example for terra-status-view.
```

File: packages/terra-status-view/src/StatusView.jsx [hunk 6]
```diff
@@ -164,7 +164,7 @@ const StatusView = ({
   return (
     <div {...customProps} className={outerViewClassNames}>
```
Comment (supreethmr) [inline packages/terra-status-view/src/StatusView.jsx:165]: can we get role and aria attribute to be added at top div to ensure that they are not overridden by custom props. !!

Comment (supreethmr) [reply packages/terra-status-view/src/StatusView.jsx:165]: nit: role should be added before the Aria
```js
role="alert" aria-live="polite"
```

**Review by supreethmr — APPROVED** [review]

**Pull request merged**